src/MessageTrait.php

```diff
@@ -262,14 +274,18 @@ public function withAddedHeader($header, $value)
      */
     public function withoutHeader($header)
     {
-        if (! $this->hasHeader($header)) {
-            return clone $this;
+        $normalized = strtolower($header);
+
+        if ($normalized === 'host') {
+            return $this;
         }
```

Comment:
This change doesn't make sense to me. The `Host` header is specific to requests, which is why that class had additional logic present for handling it. Can you explain the rationale you have in making this change, as well as the changes to `Request` and `RequestTrait`, please? I'm not following what the changes accomplish, and it looks like they could potentially affect `Response` messages as well as currently proposed.